packages/headless/src/features/product-recommendations/product-recommendations-actions.ts

```diff
@@ -102,20 +102,35 @@ export const getProductRecommendations = createAsyncThunk<
   }
 );
 
-// TODO: Review the mappings here
 const mapResultToProductResult = (result: Result): ProductRecommendation => {
+  const price = (result.raw.ec_price || result.raw.price) as number | undefined;
+  const promoPrice = (result.raw.ec_promo_price || result.raw.promo_price) as
+    | number
+    | undefined;
+  const inStock = (result.raw.ec_in_stock || result.raw.in_stock) as
+    | string
+    | boolean
+    | undefined;
```

Comment:
This is apparently meant to return a string, but I added support for boolean just in case since I couldn't get a request to include `ec_in_stock`.

Should I remove support for boolean?

As for the `undefined` case, I'd really rather not collapse it since, to me, the lack of information on whether something is in stock or not does not imply that it's not in stock or that it is, I would leave it to the UI.

Reply:
Ok, sounds good for `undefined`.  For `string` vs. `boolean`, I think let's just support `string` for now?

I guess another question is why does `ec_in_stock` have priority over `in_stock`? e.g. could we enter a situation where `ec_in_stock` is `no`, while `in_stock` is `yes`?

Reply:
I wouldn't be against exposing `result`. I could imagine a developer wanting to show more fields.

Reply:
Karen was against exposing `result` :smile: 

We wanted to have a strong opinion on what should be shown in the Recommendations, and these are the fields. We might add some in the long run, but exposing `result` will make everyone use anything

Reply:
Ok, that's fair. Since we want to be opinionated on what is exposed, could we also be opinionated on what fields should be used when indexing? Could we support only one of `ec_in_stock` or `in_stock`?

I think this will lead to less confusion and bugs, whether at index time or inside headless.

Reply:
I initially thought it would be weird to only accept the standard fields (`ec_in_stock`) and expose it as `inStock`.

So the question is, would it cause **more** confusion if only `in_stock` is defined and `inStock` didn't use it? :confused: 

(I think my answer is yes, but I'm open to discussion :smile: )

Reply:
My answer would depend on an answer to another question.

Let's assume `in_stock` is defined, and `ec_in_stock` is not. Would anything else fail? e.g. machine learning, something related to the commerce/catalogue service.

If yes, then I think the headless controller should also fail/return `undefined`. If not, then the customer could be under the assumption that everything is okay (headless controller is returning the correct value), but there are silent errors at other points in the stack.

In general, I always encourage having one way to do something. However, if the rest of the stack supports two ways, then headless should also support the same two ways.

Reply:
That is a good point! 

ML will only leverage fields that use the standard mapping (`ec_` prefix), so in this mindset, we should only map `ec_` field.

@btaillon you good with that? :smile:

Reply:
Ok, cool, should we also give the same treatment to all exposed properties? I see name returns either `ec_name` or `title`, the description is the `ec_description` or `excerpt`.

Should the commerce controllers just focus on `ec_` fields? If something is `undefined`, then this will be a sign to reindex so that ML can learn. The CartRecommendationList controller will become exclusively an e-commerce controller.

Reply:
Yes, I agree with all of this :smile: name should only use `ec_name`, and so on

Reply:
I agree too. This is something we will have to indicate in our standards field, (that the standard fields are needed for recommender models) 
We will need to think of a way to indicate to user that there are missing fields in their catalog configuration at some points too. 
I guess having them as "required" in the catalog configuration would be a solution but since there is no guaranty that the client will use the recommender maybe it's overkilled 
(but again, which eCommerce client wouldn't benefit from using recommender?)

Reply:
Alright, so what about:
* `sku`
  Should we still read it from `raw.productid`? `raw.sku`?
* `link`
  Should we still read it from `clickUri`? Otherwise, what would it be?
* `tags`
  Should we still read it at all? Otherwise, would it become `ec_tags`?

Also, should we remove `description` and `shortDescription`?

Reply:
@francoislg @kpoulincoveo opinions?

Reply:
:point_up: 

`sku` => `permanentid`. This is what will be used according to the Standard.

`link` => `clickUri`. We have no special link, so I guess we should keep this one.

`tags` => Let's just remove it

`description` => Remove, since we should not use it

`shortDescription` => I would keep this one, it might be useful for accessibility purposes :thinking: